### PR TITLE
Mineclone2 compatibility

### PR DESCRIPTION
**Pull request by minertestdude**
- ability to run also within Mineclone 2 
- custom white UI for Mineclone2, regular (unchanged) for Minetest Game

### CODE REVIEW

File: README.txt [hunk 1]
```diff
@@ -8,7 +8,9 @@
 This mod adds a wooden trash can, and a dumpster to the game.  Right click it, put in your trash, and click the empty trash button.
 You can also throw things in the wooden trash can by pressing "q" or throwing them out of your inventory.
 
-Version: 0.2.2
+This is a "P" version, based on 0.2.2 enhanced with ability to run with MineClone2 game.
+
+Version: 0.2.2-P
```
Comment (SmallJoker) [inline README.txt:13]: Just raise it to 0.2.3

File: README.txt [hunk 2]
```diff
@@ -8,7 +8,9 @@
 This mod adds a wooden trash can, and a dumpster to the game.  Right click it, put in your trash, and click the empty trash button.
 You can also throw things in the wooden trash can by pressing "q" or throwing them out of your inventory.
 
-Version: 0.2.2
+This is a "P" version, based on 0.2.2 enhanced with ability to run with MineClone2 game.
```
Comment (SmallJoker) [inline README.txt:11]: Also good to mention Minetest Game in this case: `After version 0.2.3, this mod is compatible with Minetest Game and MineClone 2`.

File: description.txt [hunk 3]
```diff
@@ -1 +1 @@
-This mod adds a wooden trash can, and a dumpster to the game.  Right click it, put in your trash, and click the empty trash button.
+This mod adds a wooden trash can, and a dumpster to the game.  Right click it, put in your trash, and click the empty trash button. It is MineTest Game and MineClone 2 compatible.
```
Comment (SmallJoker) [inline description.txt:1]: Minor but it's `Minetest Game`

File: init.lua [hunk 4]
```diff
@@ -91,9 +115,10 @@ minetest.register_node("trash_can:trash_can_wooden",{
 			"size[8,9]" ..
 			"button[0,0;2,1;empty;Empty Trash]" ..
 			"list[context;trashlist;3,1;2,3;]" ..
-			"list[current_player;main;0,5;8,4;]"
+			"list[current_player;main;0,5;8,4;]" ..
+			moditems.BOXART
 		)
-		meta:set_string("infotext", "Trash Can")
+		meta:set_string("infotext", moditems.INFOBOX_CAN)
```
Comment (SmallJoker) [inline init.lua:121]: `INFOBOX_CAN` is a table in the MCL2 version. `set_string` does not work (properly?) with tables.

File: init.lua [hunk 5]
```diff
@@ -155,17 +180,18 @@ minetest.register_node("trash_can:dumpster", {
...
 		meta:set_string("formspec",
 			"size[8,9]" ..
 			"button[0,0;2,1;empty;Empty Trash]" ..
 			"list[context;main;1,1;6,3;]" ..
-			"list[current_player;main;0,5;8,4;]"
+			"list[current_player;main;0,5;8,4;]"..
+			moditems.BOXART
 		)
-		meta:set_string("infotext", "Dumpster")
+		meta:set_string("infotext", moditems.INFOBOX_DUMP)
```
Comment (SmallJoker) [inline init.lua:194]: Is a table for MCL2 here too.

File: init.lua [hunk 6]
```diff
@@ -1,3 +1,27 @@
...
+local moditems = {}  -- switcher
+
+if core.get_modpath("mcl_core") and mcl_core then -- means MineClone 2 is loaded, this is its core mod
+	moditems.IRON_ITEM = "mcl_core:iron_ingot"   -- MCL version of iron ingot
+	moditems.COAL_ITEM = "mcl_core:coalblock" -- MCL version of coal block
+	moditems.GREEN_DYE = "mcl_dye:green" -- MCL version of green dye
+	moditems.METAL_SOUNDS = mcl_sounds.node_sound_metal_defaults()
+	moditems.INFOBOX_CAN = {}
+	moditems.INFOBOX_DUMP = {}
+	moditems.BOXART = "bgcolor[#d0d0d0;false]listcolors[#9d9d9d;#9d9d9d;#5c5c5c;#000000;#ffffff]" -- trying to imitate MCL boxart
```
Comment (SmallJoker) [inline init.lua:12]: Just asking: Isn't there a public MCL2 string which contains this information?

**Comment by minertestdude** [comment]
@SmallJoker Hi! Sorry for very late response! I've gone through your messages and created another version, which already includes the changes that you implemented meanwhile, so I will open new pull request for this. 
* The table inside set_string works properly on my side, example: https://imgur.com/a/XgEpYep  All objects in MCL2 by default appear without flying captions over them, hence the strings are absent for mcl_core.
* As of boxart, I don't think its available. I have spent time fishing this information out and don't remember where exactly I took it - it could be that I made screenshot then used GIMP tool to get color values, because the mod uses formspec and mcl2 uses own menu (which I couldn't figure out back in day), so just found similar colors.
Best

**Pull request closed without merging**